src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -328,21 +328,21 @@ private IndexedNode(int bitmap, List<AbstractNode<K, V>> subNodes) {
             int bit = mask;
             int count = 0;
             List<AbstractNode<K, V>> sub = subNodes;
-            List<AbstractNode<K, V>> arr = List.empty();
+            final Object[] arr = new Object[BUCKET_SIZE];
```

Comment:
<code>javaslang.collection.Array</code> backed by <code>Object[]</code> but not by <code>T[]</code> because of generic types. It is not possible in Java create arrays like <code>Tuple2&lt;Integer,Integer&gt;[]</code>